src/core/Context.py

```diff
@@ -3,32 +3,37 @@
 import asyncio
 import io
 from contextlib import suppress
-from typing import TYPE_CHECKING, Any, Optional, Union
+from typing import TYPE_CHECKING, Any, Callable, Generic, Optional, TypeVar, Union
 
+import aiohttp
+import config as cfg
 import discord
+import utils
 from async_property import async_property
 from discord.ext import commands
 
-import config as cfg
-import utils
+BotT = TypeVar("BotT", bound=commands.Bot)
+
 
+__all__ = ("Context",)
 
-class Context(commands.Context):
+
+class Context(commands.Context["commands.Bot"], Generic[BotT]):
```

Comment:
`Generic` class is sexy.
```py
# Helps in better code suggestion from copilot
ctx: Context[Quotient] = ...  # IDE may scream, like "expected 'BotT', 'Quotient' found". It can be ignored
```